<Type Name="SEHException" FullName="System.Runtime.InteropServices.SEHException">
  <Metadata>
    <Meta Name="ms.openlocfilehash" Value="2006f6528910fbef9742e14057a3aeaaabd43cff" />
    <Meta Name="ms.sourcegitcommit" Value="d31dc2ede16f6f7bc64e90d9f897ff54c4e3869b" />
    <Meta Name="ms.translationtype" Value="HT" />
    <Meta Name="ms.contentlocale" Value="pt-BR" />
    <Meta Name="ms.lasthandoff" Value="04/03/2018" />
    <Meta Name="ms.locfileid" Value="30479900" />
  </Metadata>
  <TypeSignature Language="C#" Value="public class SEHException : System.Runtime.InteropServices.ExternalException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi serializable beforefieldinit SEHException extends System.Runtime.InteropServices.ExternalException" />
  <TypeSignature Language="DocId" Value="T:System.Runtime.InteropServices.SEHException" />
  <TypeSignature Language="VB.NET" Value="Public Class SEHException&#xA;Inherits ExternalException" />
  <TypeSignature Language="C++ CLI" Value="public ref class SEHException : System::Runtime::InteropServices::ExternalException" />
  <AssemblyInfo>
    <AssemblyName>System.Runtime.InteropServices</AssemblyName>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
    <AssemblyVersion>4.0.10.0</AssemblyVersion>
    <AssemblyVersion>4.0.20.0</AssemblyVersion>
    <AssemblyVersion>4.1.0.0</AssemblyVersion>
    <AssemblyVersion>4.1.1.0</AssemblyVersion>
    <AssemblyVersion>4.2.0.0</AssemblyVersion>
    <AssemblyVersion>4.2.1.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>mscorlib</AssemblyName>
    <AssemblyVersion>2.0.5.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <AssemblyInfo>
    <AssemblyName>netstandard</AssemblyName>
    <AssemblyVersion>2.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.Exception</BaseTypeName>
    <BaseTypeName FrameworkAlternate="netframework-4.5">System.Runtime.InteropServices.ExternalException</BaseTypeName>
  </Base>
  <Interfaces />
  <Attributes>
    <Attribute>
      <AttributeName>System.Runtime.InteropServices.ComVisible(true)</AttributeName>
    </Attribute>
  </Attributes>
  <Docs>
    <summary>
      <span data-ttu-id="326e9-101">Representa erros de SEH (Manipulação de Exceção Estruturada).</span>
      <span class="sxs-lookup">
        <span data-stu-id="326e9-101">Represents structured exception handling (SEH) errors.</span>
      </span>
    </summary>
    <remarks>
      <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="326e9-102">O <xref:System.Runtime.InteropServices.SEHException> classe manipula erros de SEH gerados do código não gerenciado, mas que não foram mapeados para outra exceção do .NET Framework.</span><span class="sxs-lookup"><span data-stu-id="326e9-102">The <xref:System.Runtime.InteropServices.SEHException> class handles SEH errors that are thrown from unmanaged code, but that have not been mapped to another .NET Framework exception.</span></span> <span data-ttu-id="326e9-103">O <xref:System.Runtime.InteropServices.SEHException> classe também corresponde do `HRESULT` `E_FAIL` (0x80004005).</span><span class="sxs-lookup"><span data-stu-id="326e9-103">The <xref:System.Runtime.InteropServices.SEHException> class also corresponds to the `HRESULT` `E_FAIL` (0x80004005).</span></span>  
  
 <span data-ttu-id="326e9-104">Geralmente, o .NET Framework encontra exceções SEH não gerenciadas que são mapeadas automaticamente para equivalentes gerenciados.</span><span class="sxs-lookup"><span data-stu-id="326e9-104">The .NET Framework often encounters unmanaged SEH exceptions that are automatically mapped to managed equivalents.</span></span> <span data-ttu-id="326e9-105">Há duas exceções SEH comuns não gerenciadas:</span><span class="sxs-lookup"><span data-stu-id="326e9-105">There are two common unmanaged SEH exceptions:</span></span>  
  
-   <span data-ttu-id="326e9-106">`STATUS_NO_MEMORY` exceções são mapeadas automaticamente para o <xref:System.OutOfMemoryException> classe.</span><span class="sxs-lookup"><span data-stu-id="326e9-106">`STATUS_NO_MEMORY` exceptions are automatically mapped to the <xref:System.OutOfMemoryException> class.</span></span>  
  
-   <span data-ttu-id="326e9-107">`STATUS_ACCESS_VIOLATION` exceções automaticamente são mapeadas da seguinte maneira:</span><span class="sxs-lookup"><span data-stu-id="326e9-107">`STATUS_ACCESS_VIOLATION` exceptions are automatically mapped as follows:</span></span>  
  
    -   <span data-ttu-id="326e9-108">Se `legacyNullReferencePolicy` é aplicada, todas as violações de acesso são mapeadas para o <xref:System.NullReferenceException> classe.</span><span class="sxs-lookup"><span data-stu-id="326e9-108">If `legacyNullReferencePolicy` is applied, all access violations are mapped to the <xref:System.NullReferenceException> class.</span></span>  
  
    -   <span data-ttu-id="326e9-109">Se o endereço no qual foi tentada a leitura/gravação não está em código com compilação JIT, a exceção será mapeada para o <xref:System.AccessViolationException> classe.</span><span class="sxs-lookup"><span data-stu-id="326e9-109">If the address at which the read/write was attempted is not in JIT-compiled code, the exception is mapped to the <xref:System.AccessViolationException> class.</span></span>  
  
    -   <span data-ttu-id="326e9-110">Se o endereço no qual foi tentada a leitura/gravação está em código com compilação JIT, mas ele não está na área de partição de SO Null, a exceção será mapeada para o <xref:System.AccessViolationException> classe.</span><span class="sxs-lookup"><span data-stu-id="326e9-110">If the address at which the read/write was attempted is in JIT-compiled code, but it is not in the OS Null partition area, the exception is mapped to the <xref:System.AccessViolationException> class.</span></span>  
  
    -   <span data-ttu-id="326e9-111">Se não houver nenhum `legacyNullReferencePolicy`e o endereço no qual foi tentada a leitura/gravação está no código com compilação JIT e na área de partição do sistema operacional Null, a exceção será mapeada para o <xref:System.NullReferenceException> classe.</span><span class="sxs-lookup"><span data-stu-id="326e9-111">If there is no `legacyNullReferencePolicy`, and the address at which the read/write was attempted is in JIT-compiled code and in the OS Null partition area, the exception is mapped to the <xref:System.NullReferenceException> class.</span></span>  
  
 <span data-ttu-id="326e9-112">Exceção SEH automaticamente não está mapeada para uma exceção específica é mapeada para o <xref:System.Runtime.InteropServices.SEHException> classe por padrão.</span><span class="sxs-lookup"><span data-stu-id="326e9-112">Any SEH exception that is not automatically mapped to a specific exception is mapped to the <xref:System.Runtime.InteropServices.SEHException> class by default.</span></span>  
  
 <span data-ttu-id="326e9-113">Para obter mais informações, pesquise "exceções não gerenciadas" e "Estruturado tratamento de exceção" na biblioteca MSDN.</span><span class="sxs-lookup"><span data-stu-id="326e9-113">For more information, search on "unmanaged exceptions" and "Structured Exception Handling" in the MSDN Library.</span></span>  
  
 <span data-ttu-id="326e9-114">Observe que o <xref:System.Runtime.InteropServices.SEHException> classe não causa destruidores de exceção de C++ não gerenciados a ser chamado.</span><span class="sxs-lookup"><span data-stu-id="326e9-114">Note that the <xref:System.Runtime.InteropServices.SEHException> class does not cause unmanaged C++ exception destructors to be called.</span></span> <span data-ttu-id="326e9-115">Para garantir que são chamados de destruidores de exceção de C++ não gerenciados, use a seguinte sintaxe no `catch` bloco.</span><span class="sxs-lookup"><span data-stu-id="326e9-115">To ensure that unmanaged C++ exception destructors are called, use the following syntax in the `catch` block.</span></span>  
  
```vb  
Catch   
     ' Handle catch here.  
End Try  
```  
  
```csharp  
catch  
{  
     // Handle catch here.  
}  
```  
  
```cpp  
catch(…)  
{  
     // Handle catch here.  
}   
```  
  
 ]]></format>
    </remarks>
    <altmember cref="T:System.Exception" />
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>System.Runtime.InteropServices</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary>
          <span data-ttu-id="326e9-116">Inicializa uma nova instância da classe <see cref="T:System.Runtime.InteropServices.SEHException" />.</span>
          <span class="sxs-lookup">
            <span data-stu-id="326e9-116">Initializes a new instance of the <see cref="T:System.Runtime.InteropServices.SEHException" /> class.</span>
          </span>
        </summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public SEHException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Runtime.InteropServices.SEHException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; SEHException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime.InteropServices</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary>
          <span data-ttu-id="326e9-117">Inicializa uma nova instância da classe <see cref="T:System.Runtime.InteropServices.SEHException" />.</span>
          <span class="sxs-lookup">
            <span data-stu-id="326e9-117">Initializes a new instance of the <see cref="T:System.Runtime.InteropServices.SEHException" /> class.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="326e9-118">O <xref:System.Runtime.InteropServices.SEHException> classe herda o <xref:System.Runtime.InteropServices.ExternalException> classe.</span><span class="sxs-lookup"><span data-stu-id="326e9-118">The <xref:System.Runtime.InteropServices.SEHException> class inherits from the <xref:System.Runtime.InteropServices.ExternalException> class.</span></span> <span data-ttu-id="326e9-119">Este construtor define as propriedades do <xref:System.Exception> objeto com os valores na tabela a seguir.</span><span class="sxs-lookup"><span data-stu-id="326e9-119">This constructor sets the properties of the <xref:System.Exception> object to the values in the following table.</span></span>  
  
|<span data-ttu-id="326e9-120">Propriedade</span><span class="sxs-lookup"><span data-stu-id="326e9-120">Property</span></span>|<span data-ttu-id="326e9-121">Valor</span><span class="sxs-lookup"><span data-stu-id="326e9-121">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="326e9-122">Uma referência nula (`Nothing` no Visual Basic).</span><span class="sxs-lookup"><span data-stu-id="326e9-122">A null reference (`Nothing` in Visual Basic).</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="326e9-123">Uma cadeia de caracteres de mensagem de erro localizada.</span><span class="sxs-lookup"><span data-stu-id="326e9-123">A localized error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public SEHException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Runtime.InteropServices.SEHException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; SEHException(System::String ^ message);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime.InteropServices</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="326e9-124">A mensagem que indica o motivo da exceção.</span>
          <span class="sxs-lookup">
            <span data-stu-id="326e9-124">The message that indicates the reason for the exception.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="326e9-125">Inicializa uma nova instância da classe <see cref="T:System.Runtime.InteropServices.SEHException" /> com uma mensagem especificada.</span>
          <span class="sxs-lookup">
            <span data-stu-id="326e9-125">Initializes a new instance of the <see cref="T:System.Runtime.InteropServices.SEHException" /> class with a specified message.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="326e9-126"><xref:System.Runtime.InteropServices.SEHException> herda de <xref:System.Runtime.InteropServices.ExternalException>.</span><span class="sxs-lookup"><span data-stu-id="326e9-126"><xref:System.Runtime.InteropServices.SEHException> inherits from <xref:System.Runtime.InteropServices.ExternalException>.</span></span> <span data-ttu-id="326e9-127">Este construtor define as propriedades do <xref:System.Exception> objeto conforme mostrado na tabela a seguir.</span><span class="sxs-lookup"><span data-stu-id="326e9-127">This constructor sets the properties of the <xref:System.Exception> object as shown in the following table.</span></span>  
  
|<span data-ttu-id="326e9-128">Propriedade</span><span class="sxs-lookup"><span data-stu-id="326e9-128">Property</span></span>|<span data-ttu-id="326e9-129">Valor</span><span class="sxs-lookup"><span data-stu-id="326e9-129">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="326e9-130">Uma referência nula (`Nothing` no Visual Basic).</span><span class="sxs-lookup"><span data-stu-id="326e9-130">A null reference (`Nothing` in Visual Basic).</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="326e9-131">`message`.</span><span class="sxs-lookup"><span data-stu-id="326e9-131">`message`.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="protected SEHException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);" />
      <MemberSignature Language="ILAsm" Value=".method familyhidebysig specialname rtspecialname instance void .ctor(class System.Runtime.Serialization.SerializationInfo info, valuetype System.Runtime.Serialization.StreamingContext context) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Runtime.InteropServices.SEHException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)" />
      <MemberSignature Language="VB.NET" Value="Protected Sub New (info As SerializationInfo, context As StreamingContext)" />
      <MemberSignature Language="C++ CLI" Value="protected:&#xA; SEHException(System::Runtime::Serialization::SerializationInfo ^ info, System::Runtime::Serialization::StreamingContext context);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>System.Runtime.InteropServices</AssemblyName>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="info" Type="System.Runtime.Serialization.SerializationInfo" />
        <Parameter Name="context" Type="System.Runtime.Serialization.StreamingContext" />
      </Parameters>
      <Docs>
        <param name="info">
          <span data-ttu-id="326e9-132">O objeto que mantém os dados de objeto serializados.</span>
          <span class="sxs-lookup">
            <span data-stu-id="326e9-132">The object that holds the serialized object data.</span>
          </span>
        </param>
        <param name="context">
          <span data-ttu-id="326e9-133">As informações contextuais sobre a origem ou o destino.</span>
          <span class="sxs-lookup">
            <span data-stu-id="326e9-133">The contextual information about the source or destination.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="326e9-134">Inicializa uma nova instância da classe <see cref="T:System.Runtime.InteropServices.SEHException" /> dos dados de serialização.</span>
          <span class="sxs-lookup">
            <span data-stu-id="326e9-134">Initializes a new instance of the <see cref="T:System.Runtime.InteropServices.SEHException" /> class from serialization data.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="326e9-135">Este construtor é chamado durante a desserialização para reconstituir o objeto de exceção transmitido por um fluxo.</span><span class="sxs-lookup"><span data-stu-id="326e9-135">This constructor is called during deserialization to reconstitute the exception object transmitted over a stream.</span></span> <span data-ttu-id="326e9-136">Para obter mais informações, consulte o namespace de <xref:System.Runtime.Serialization>.</span><span class="sxs-lookup"><span data-stu-id="326e9-136">For more information, see the <xref:System.Runtime.Serialization> namespace.</span></span>  
  
 <span data-ttu-id="326e9-137"><xref:System.Runtime.InteropServices.SEHException> herda de <xref:System.Runtime.InteropServices.ExternalException>.</span><span class="sxs-lookup"><span data-stu-id="326e9-137"><xref:System.Runtime.InteropServices.SEHException> inherits from <xref:System.Runtime.InteropServices.ExternalException>.</span></span>  
  
 ]]></format>
        </remarks>
        <exception cref="T:System.ArgumentNullException">
          <span data-ttu-id="326e9-138">
            <paramref name="info" /> é <see langword="null" />.</span>
          <span class="sxs-lookup">
            <span data-stu-id="326e9-138">
              <paramref name="info" /> is <see langword="null" />.</span>
          </span>
        </exception>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public SEHException (string message, Exception inner);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception inner) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Runtime.InteropServices.SEHException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, inner As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; SEHException(System::String ^ message, Exception ^ inner);" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime.InteropServices</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="inner" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="326e9-139">A mensagem de erro que explica a razão da exceção.</span>
          <span class="sxs-lookup">
            <span data-stu-id="326e9-139">The error message that explains the reason for the exception.</span>
          </span>
        </param>
        <param name="inner">
          <span data-ttu-id="326e9-140">A exceção que é a causa da exceção atual.</span>
          <span class="sxs-lookup">
            <span data-stu-id="326e9-140">The exception that is the cause of the current exception.</span>
          </span>
          <span data-ttu-id="326e9-141">Se o parâmetro <c>inner</c> não for <see langword="null" />, a exceção atual será gerada em um bloco <see langword="catch" /> que manipula a exceção interna.</span>
          <span class="sxs-lookup">
            <span data-stu-id="326e9-141">If the <c>inner</c> parameter is not <see langword="null" />, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="326e9-142">Inicializa uma nova instância da classe <see cref="T:System.Runtime.InteropServices.SEHException" /> com uma mensagem de erro especificada e uma referência à exceção interna que é a causa da exceção.</span>
          <span class="sxs-lookup">
            <span data-stu-id="326e9-142">Initializes a new instance of the <see cref="T:System.Runtime.InteropServices.SEHException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="326e9-143">Uma exceção que é lançada como um resultado direto de uma exceção anterior deve incluir uma referência para a exceção anterior na propriedade <xref:System.Exception.InnerException%2A>.</span><span class="sxs-lookup"><span data-stu-id="326e9-143">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="326e9-144">A propriedade <xref:System.Exception.InnerException%2A> retorna o mesmo valor que é passado para o construtor, ou `null` caso a propriedade <xref:System.Exception.InnerException%2A> não forneça o valor de exceção interno para o construtor.</span><span class="sxs-lookup"><span data-stu-id="326e9-144">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or `null` if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="326e9-145">A tabela a seguir mostra os valores de propriedade iniciais para uma instância de <xref:System.Runtime.InteropServices.SEHException>.</span><span class="sxs-lookup"><span data-stu-id="326e9-145">The following table shows the initial property values for an instance of <xref:System.Runtime.InteropServices.SEHException>.</span></span>  
  
|<span data-ttu-id="326e9-146">Propriedade</span><span class="sxs-lookup"><span data-stu-id="326e9-146">Property</span></span>|<span data-ttu-id="326e9-147">Valor</span><span class="sxs-lookup"><span data-stu-id="326e9-147">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="326e9-148">A referência de exceção interna.</span><span class="sxs-lookup"><span data-stu-id="326e9-148">The inner exception reference.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="326e9-149">A cadeia de caracteres da mensagem de erro.</span><span class="sxs-lookup"><span data-stu-id="326e9-149">The error message string.</span></span>|  
  
 ]]></format>
        </remarks>
        <altmember cref="T:System.Exception" />
      </Docs>
    </Member>
    <Member MemberName="CanResume">
      <MemberSignature Language="C#" Value="public virtual bool CanResume ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig newslot virtual instance bool CanResume() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Runtime.InteropServices.SEHException.CanResume" />
      <MemberSignature Language="VB.NET" Value="Public Overridable Function CanResume () As Boolean" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; virtual bool CanResume();" />
      <MemberType>Method</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Runtime.InteropServices</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.10.0</AssemblyVersion>
        <AssemblyVersion>4.0.20.0</AssemblyVersion>
        <AssemblyVersion>4.1.0.0</AssemblyVersion>
        <AssemblyVersion>4.1.1.0</AssemblyVersion>
        <AssemblyVersion>4.2.0.0</AssemblyVersion>
        <AssemblyVersion>4.2.1.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>mscorlib</AssemblyName>
        <AssemblyVersion>2.0.5.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <AssemblyInfo>
        <AssemblyName>netstandard</AssemblyName>
        <AssemblyVersion>2.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <ReturnValue>
        <ReturnType>System.Boolean</ReturnType>
      </ReturnValue>
      <Parameters />
      <Docs>
        <summary>
          <span data-ttu-id="326e9-150">Indica se a exceção pode ser recuperada e se o código pode continuar do ponto no qual a exceção foi gerada.</span>
          <span class="sxs-lookup">
            <span data-stu-id="326e9-150">Indicates whether the exception can be recovered from, and whether the code can continue from the point at which the exception was thrown.</span>
          </span>
        </summary>
        <returns>
          <span data-ttu-id="326e9-151">Sempre <see langword="false" />, pois exceções retomáveis não são implementadas.</span>
          <span class="sxs-lookup">
            <span data-stu-id="326e9-151">Always <see langword="false" />, because resumable exceptions are not implemented.</span>
          </span>
        </returns>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="326e9-152">Se <xref:System.Runtime.InteropServices.SEHException.CanResume%2A> retorna `true`, em seguida, um manipulador de exceção filtrados pode corrigir o problema que causou a exceção e o código continuará a partir do ponto em que a exceção foi lançada.</span><span class="sxs-lookup"><span data-stu-id="326e9-152">If <xref:System.Runtime.InteropServices.SEHException.CanResume%2A> returns `true`, then a filtered exception handler can correct the problem that caused the exception, and the code will continue from the point at which the exception was thrown.</span></span>  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
  </Members>
</Type>